Umee-Bots/Stale-Price/src/agent.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  BlockEvent,
+  ethers,
+  Finding,
+  getEthersProvider,
+  HandleBlock,
+  HandleTransaction,
+  Initialize,
+  TransactionEvent,
+} from "forta-agent";
+
+import CONFIG from "./agent.config";
+
+import utils, { AgentConfig, AssetDataI } from "./utils";
+
+const assetsDataList: AssetDataI[] = [];
+
+export const provideInitialize =
+  (provider: ethers.providers.Provider): Initialize =>
+  async () => {
+    assetsDataList.push(...(await utils.getAssetData(CONFIG, provider)));
+  };
+
+export const provideHandleTransaction = (
+  config: AgentConfig,
+  provider: ethers.providers.Provider,
+  assetsDataList: AssetDataI[]
+): HandleTransaction => {
+  const handleTransaction: HandleTransaction = async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+    const updateSourceLogs = txEvent.filterLog(utils.EVENT_ABI, config.umeeOracleAddress);
+    await Promise.all(
+      updateSourceLogs.map(async (logs) => {
+        const [asset, source] = logs.args;
+        const assetsDataIndex = assetsDataList.findIndex((assetsData) => {
+          return assetsData.asset;
+        });
```

Comment:
```suggestion
        const assetsDataIndex = assetsDataList.findIndex((assetsData) => {
          return assetsData.asset === asset.toLowerCase();
        });
```
Updating or adding asset data entries is something a specific unit test should be made for. In this case, `findIndex` was simply returning `-1` if the array was empty and `0` otherwise, since `assetsData.asset` is always truthy.